src/components/PopupWithForm.js

```diff
@@ -4,27 +4,33 @@ class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
     super({ popupSelector });
     this._popupForm = this._modalEl.querySelector(".modal__form");
+    this._submitButton = this._modalEl.querySelector(".modal__save-button");
     this._handleFormSubmit = handleFormSubmit;
     this._list = [...this._popupForm.querySelectorAll(".modal__input")];
   }
 
+  open() {
+    this._submitButton.textContent = "Save";
+    super.open();
+  }
+
   close() {
     this._popupForm.reset();
     super.close();
   }
 
-  // setInputValues(data) {
-  //   this._inputList.forEach((input) => {
-  //     input.value = data[input.name];
-  //   });
-  // }
+  setLoading(loading) {
+    if (loading) {
+      this._submitButton.textContent = "Saving...";
+    }
+  }
 
   _getInputValues() {
-    // this.setInputValues();
     const inputData = {};
     this._list.forEach((input) => {
       inputData[input.name] = input.value;
     });
+
     return inputData;
   }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`